src/test/regress/specs/isolation_create_table_vs_add_remove_node.spec

```diff
@@ -0,0 +1,108 @@
+setup
+{
+	SELECT 1 FROM master_add_node('localhost', 57637);
+	SELECT * FROM master_get_active_worker_nodes() ORDER BY node_name, node_port;
+}
+
+teardown
+{
+	DROP TABLE IF EXISTS dist_table;
+
+	SELECT master_remove_node(nodename, nodeport) FROM pg_dist_node;
+}
+
+session "s1"
+
+step "s1-begin"
+{
+	BEGIN;
+}
+
+step "s1-add-node-2"
+{
+	SELECT 1 FROM master_add_node('localhost', 57638);
+}
+
+step "s1-remove-node-2"
+{
+	SELECT * FROM master_remove_node('localhost', 57638);
+}
+
+step "s1-abort"
+{
+	ABORT;
+}
+
+step "s1-commit"
+{
+	COMMIT;
+}
+
+step "s1-query-table"
+{
+	SELECT * FROM dist_table;
+}
+
+step "s1-show-nodes"
+{
+	SELECT nodename, nodeport, isactive FROM pg_dist_node ORDER BY nodename, nodeport;
+}
+
+step "s1-show-placements"
+{
+	SELECT
+		nodename, nodeport
+	FROM
+		pg_dist_shard_placement JOIN pg_dist_shard USING (shardid)
+	WHERE
+		logicalrelid = 'dist_table'::regclass
+	ORDER BY
+		nodename, nodeport;
+}
+
+session "s2"
+
+step "s2-begin"
+{
+	BEGIN;
+}
+
+step "s2-create-table-1"
+{
+	SET citus.shard_count TO 4;
+	SET citus.shard_replication_factor TO 1;
+	CREATE TABLE dist_table (x int, y int);
+	SELECT create_distributed_table('dist_table', 'x');
+}
+
+step "s2-create-table-2"
```

Comment:
we should also have append version of this. I added a failing test case to comments